compiler/modules/front/codeGen.bal

```diff
@@ -28,9 +28,16 @@ type Binding record {|
     Binding? unnarrowed = ();
 |};
 
+type StmtNarrowing record {|
+    Binding binding;
+    t:SemType ifCompletesNormally;
+    bir:Result basis;
+|};
+
 type StmtEffect record {|
     bir:BasicBlock? block;
     Binding? bindings = ();
+    StmtNarrowing[] narrowings = [];
```

Comment:
I think it should work to store the basis in the binding, instead of doing this.